mate-panel/panel-run-dialog.c

```diff
@@ -729,18 +754,26 @@ panel_run_dialog_find_command_idle (PanelRunDialog *dialog)
 			gtk_list_store_set (dialog->program_list_store,
 					    &iter,
 					    COLUMN_VISIBLE, TRUE,
+					    COLUMN_ACCELERATOR_MASK, (gint)accelerator_key_mapping[visible_program_idx].modifier,
+					    COLUMN_ACCELERATOR_KEY_VALUE, accelerator_key_mapping[visible_program_idx].key_id,
 					    -1);
+                        if (accelerator_key_mapping[visible_program_idx].list_item_idx != -1) visible_program_idx++;
```

Comment:
Doesn't this mean all entries after the 9th will have the 10th accelerator?  I don't see anything that prevents setting the accelerator if it's past the last one.

I think I would actually move this outside those tests, and just have a flag "this is visible", and depending on that set or not the accel.  Something like this pseudo-code, although possibly optimizing the visibility check with a local variable (not that it matters that much though):
```c
if (visible_program_idx < G_N_ELEMENTS(accelerator_key_mapping)) {
	gboolean visible = FALSE;

	gtk_tree_model_get(model, &iter, COLUMN_VISIBLE, &visible, -1);
	if (visible) {
		gtk_list_store_set (dialog->program_list_store,
				    &iter,
				    COLUMN_ACCELERATOR_MASK, (gint)accelerator_key_mapping[visible_program_idx].modifier,
				    COLUMN_ACCELERATOR_KEY_VALUE, accelerator_key_mapping[visible_program_idx].key_id,
				    -1);
		visible_program_idx++;
	}
}

Reply:
Have a look again, I tried to fix it like you suggested. The entries after the last one have the "deactivated" shortcut label.